lib/fswatcher/fswatcher.go

```diff
@@ -108,14 +115,23 @@ func (watcher *FsWatcher) watchFilesystem() {
 	}
 }
 
-func (watcher *FsWatcher) newFsEvent(eventPath string) *FsEvent {
-	if isSubpath(eventPath, watcher.folderPath) {
+func (watcher *FsWatcher) newFsEvent(eventPath string) {
+	if len(watcher.fsEvents) == maxFiles {
+		watcher.debugf("Tracking too many events; dropping: %s\n", eventPath)
```

Comment:
Shouldn't we schedule a full scan or something like that in this case, and reset the amount of evens we've queued up?